docs/nomenclature.md

```diff
@@ -18,25 +27,81 @@ Mutant currently supports the following subjects:
 * Instance methods
 * Singleton (class) methods
 
-Other subjects (constants, class bodies for DSLs, ...) are possible but aren't
-implemented in the OSS version.
+Other subjects are possible (even project-specific subjects) but aren't implemented in the OSS version. Please get in touch with the authors. Some examples are:
+* Constants
+* Class bodies for DSLs
+
+The more subjects that mutant can alter in your project, the more mutations it can create, and so the higher your confidence you can have that your tests cover the semantics of your application. Please get in touch if you require subjects beyond those implemented in Mutant already - support may be available in the commercial version.
+
+## Mutation Operator
+
+A transformation applied to the AST of a subject. Mutant knows the following high level operator classes (you will not be tested on your knowledge of these terms)
+
+Please be aware that there is no reason to learn these terms (there’s no quiz at the end of this semester!) - we’re just providing them to give you some ideas of what high-level classes of changes mutant makes to your code. For a full list of all the Mutation Operators in mutant, see the code in the [meta directory](https://github.com/mbj/mutant/tree/master/meta).
+
+* **Semantic Reduction**
+
+    This type of transformation replaces a piece of code which has somewhat complex semantics with one that has simpler semantics. To aid understanding, here are a couple of different sub-categories you could put them into. 
+    
+    * **Method call replacement** - for example, `#==` -> `#eql?` -> `#equal?`
+
+        `#==` commonly performs conversion between types in addition to checking equality, while `#eql?`  tends to check only that the class and instance variables are equal. Therefore we would say that #== is semantically simpler.
+        
+        You could also think of a semantic reduction in these cases as an increase in “strictness” of the code. `#equal?` is a stricter equality test than `#eql?`, and so on
+        
+    * **Code removal** - For example, `def my_method; do_something; end` -> `def my_method; end`
+     
+        We could argue[^1] that this category includes all of the following:
+        
+        *  Removing a particular expression, from the AST entirely
+        *  Replacing an expression with `nil`, or `true`, or `false`, or other simple literals.
```

Comment:
I don't tend to think of those as quite the same as "code removal" but maybe it belongs here

Reply:
Mm. That's true, it's more of an extreme semantic reduction, I suppose 😛

I think one thing that's missing from this whole section is a motivation for choosing to categorise things, as well as a motivation for these specific categories. I'm mostly using it as a way to structure the documentation to break it up into smaller more digestible pieces.

In my experience, when I run mutant on a subject and it reduces 2-3 lines of code to `nil` or `self`, I'm thinking "ah I just haven't specified those features in my tests", or "perhaps I could restructure that into its own method and test it individually". When I run it and it removes or reduces a smaller amount of code, I've found it's more likely that it's actually removing / reducing something that isn't necessary. This may just be a symptom of the fact my mutation testing has so far been limited to things I have built since setting up mutant.

I realise I am coming to this from a bit of a beginner's standpoint, which is perhaps not helping me to write this document in a universally applicable way. I'm happy to continue a discussion on this and to defer to experts!

Reply:
I've reworked this a bit. As well as combining the mutation and mutation operator sections, I've reworked this section a bit to add in some of that motivation. Let me know what you think of it now!